### PR TITLE
[AUD-1383] Add support for referrals-verified challenge

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

[AUD-1383](https://linear.app/audius/issue/AUD-1383/mobile-new-ui-for-verified-referral-challenge)

- Consume optimistic challenges in mobile
- Use `totalAmount` instead of `amount`, which has the rolled up aggregate challenge amounts
- Add share to twitter button
- Change copy for copy button

Client Reference:
- https://github.com/AudiusProject/audius-client/pull/821

Simulator Screen Shot - iPhone 13 - 2022-02-04 at 16 53 26
Simulator Screen Shot - iPhone 13 - 2022-02-04 at 16 59 37

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

- The gradient in the screenshots won't match what it looks like in prod - I think I still have some lingering changes in my `node-modules/react-native-linear-gradient`. Going to try and fork that repo this weekend to fix these issues.
- Not consuming amount in config strings
- Added semi-transparent black as the button underlay color for the twitter button

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

- Verified UI against Figma for both referral challenges

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

**Comment by rickyrombo** [comment]
AUD-1383

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx [hunk 1]
```diff
@@ -114,11 +119,20 @@ const createStyles = (themeColors: ThemeColors) =>
...
+      marginBottom: 12,
+      marginRight: 10
+    },
     task: {
       width: '100%',
       padding: 24,
       paddingTop: 0
     },
+    taskHeaderVerified: {
+      display: 'flex',
```
Comment (sliptype) [inline src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx:132]: FYI the default display in react-native is `flex` so this isn't needed

Also we have a `...flexRowCentered()` util that you can use if you want!

File: src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx [hunk 2]
```diff
@@ -285,13 +309,13 @@ export const ChallengeRewardsDrawer = ({
...
               styles.statusCell,
-              isComplete ? styles.statusCellComplete : {}
+              hasCompleted ? styles.statusCellComplete : {}
             ]}
           >
             <Text
               style={[
                 styles.subheader,
-                isComplete ? styles.statusTextComplete : {},
+                hasCompleted ? styles.statusTextComplete : {},
```
Comment (sliptype) [inline src/components/challenge-rewards-drawer/ChallengeRewardsDrawer.tsx:318]: You can do `hasCompleted && styles.statusTextComplete`, but that kind of falls into the larger discussion around ternary vs boolean operators

**Review by sliptype — APPROVED** [review]
Looks great!

**Pull request merged**